src/util/union_find.h

```diff
@@ -64,7 +65,7 @@ class unsigned_union_find
   void resize(size_type size)
   {
     // We only enlarge. Shrinking is yet to be implemented.
-    assert(nodes.size()<=size);
+    PRECONDITION(nodes.size() <= size);
```

Comment:
Maybe we should just change the API to only permit increasing size by a non-negative number of elements? This code has been around for a while, and "yet to be implemented" never materialised - so really it was never required.

Reply:
Yes, but what about changing the API to make this check unnecessary?

Reply:
There are two problems with an enlarge(size_t n) method: a) C++ STL usually uses resize(); b) you'd need another check like that to prevent overflowing of current_size+n.

Reply:
Indeed. Then we should probably change this into
```
if(nodes.size() > size)
  UNIMPLEMENTED;
```
to convey what the comment says.

Reply:
@tautschnig The Invariant documentation suggests that `UNIMPLEMENTED` should not be used though?